.github/workflows/debos.yml

```diff
@@ -97,3 +99,18 @@ jobs:
           EOF
           ) | sudo incus exec "${INCUS_NAME}" -- sh
 
+      - name: Upload artifacts to fileserver
+        run: |
+          set -x
+          # curl will be used to talk to fileserver; should be installed by
+          # default
+          sudo apt -y install curl
+          # create a directory for the current run
+          dir="${FILESERVER_DIR}/${GITHUB_RUN_ID}"
+          mkdir -vp "${dir}"
+          # copy output files
+          cp -v disk.img "${dir}"
+          # instruct fileserver to publish this directory
+          url="${FILESERVER_URL}/${GITHUB_RUN_ID}/"
```

Comment:
GITHUB_RUN_ID is unique within the repo. We seem to publish builds from this repo at the same level as builds from meta-qcom. that could clash no?

Reply:
Ah indeed, I assumed the run ids were globally unique, but the doc says unique per repo.

There's also the case for reruns, so a run id is actually not a unique build id; so we should also use GITHUB_RUN_ATTEMPT to have unique build ids.

Reply:
There is no globally unique run id provided by GitHub, so some repo id needs to be used. There is a repository id as a number, and there is the repository name which is the nicest IMO, but it's controller by the submitter of pull request.

I've found some non-official repositories outlining the GitHub user/org naming rules, and these explanations in the GitHub docs about user names:
https://docs.github.com/en/enterprise-cloud@latest/admin/managing-iam/iam-configuration-reference/username-considerations-for-external-authentication

Basically usernames are alphanumeric and can contain hyphens and underscores, but never two hyphens or start with an hyphen, so that should be safe to use in directory names and URLs.

Reply:
@doanac , FYI.

Reply:
ugh. i think this is going to wind up forcing a change in backend logic. the code is treating it as unique per github org and not repository. I think we have move to a format:
  `${FILESERVER_URL}/${GITHUB_REPOSITORY}/${GITHUB_RUN_ID}/`

I'll need to deploy something that's temporarily backward compatible while I work through all the places we do uploads.

Reply:
n/m - this works well. i like it.